api/filters/replacement/replacement_test.go

```diff
@@ -0,0 +1,342 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package replacement
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	filtertest "sigs.k8s.io/kustomize/api/testutils/filtertest"
+	"sigs.k8s.io/yaml"
+)
+
+func TestFilter(t *testing.T) {
+	testCases := map[string]struct {
+		input        string
+		replacements string
+		expected     string
+		expectedErr  bool
+	}{
+		"simple": {
+			input: `apiVersion: v1
+kind: Deployment
+metadata:
+  name: deploy1
+spec:
+  template:
+    spec:
+      containers:
+      - image: nginx:1.7.9
+        name: nginx-tagged
+      - image: postgres:1.8.0
+        name: postgresdb
+---
+apiVersion: v1
+kind: Deployment
+metadata:
+  name: deploy2
+spec:
+  template:
+    spec:
+      containers:
+      - image: nginx:1.7.9
+        name: nginx-tagged
+      - image: postgres:1.8.0
+        name: postgresdb
+`,
+			replacements: `replacements:
+- sourceRaw: postgres:latest
```

Comment:
Looking at these examples of what using the schema looks like in practice, I have a few questions/suggestions:
- WDYT about making a single `source` top-level field with `field` and `raw` (or maybe `value`) subfields rather than two related top-level fields with "source" in the name?
-  How about "include"/"exclude" or "select"/"reject" instead of "allow"/"deny"? The permission angle sounds a bit weird to me in a selection context.
- Since the source has to narrow down to one specific resource, is allow/deny actually useful in that context, or is it just adding complexity?

Reply:
resolving this conversation because discussion is occurring below. I think my most recent push addresses most of these concerns